ruler-models/src/commonMain/kotlin/com/spotify/ruler/models/FileType.kt

```diff
@@ -16,11 +16,57 @@
 
 package com.spotify.ruler.models
 
+import kotlinx.serialization.Serializable
+
 /** Type of an [AppFile]. */
-enum class FileType {
-    CLASS,
-    RESOURCE,
-    ASSET,
-    NATIVE_LIB,
-    OTHER,
+@Serializable
+sealed class FileType(val index: Int) {
```

Comment:
@simonschiller  I was having doubts about what would be the best way to solve this, I converted the enum to sealed class as I thought it will require the least amount of change where it is used. Let me know if you have any other ideas.

Reply:
Instead of trying to encode that information into `FileType`, what do you think about introducing another `ResourceType` enum and adding that (optionally) to the `AppFile`:
```kotlin
data class AppFile(
    ...
    val resourceType: ResourceType? = null,
) : Measurable
```

We'd only set that if the `FileType` is `RESOURCE`. That way most of the handling code could stay the same and we should even stay backwards-compatible to older JSON reports. Also if the field is `null` it will be omitted from the JSON, so the JSON report should still look clean (that's what we do for the `owner`).

Reply:
In general, I'm not a big fan of using optional null values and expect them to be not null if something else is true (in this Case FileType is Resource). That being said I do agree with the other points you've mentioned. Updated the PR 🙌

Reply:
Yeah, totally agree with that in general 😄